### ISSUE TITLE
site_admin's ability to edit/create users causes permissions problems

### ISSUE THREAD

**Issue by zerocrates** [omeka/omeka-s#219]
In addition to global_admin, site_admin has the permission to create and update users.

This causes a few problems:
- They can edit the global admins
- They can create new users with global admin privs and just use those new users
- They too would need the "self assertion" role fix Ken's working on for global admins (a triviality, but just a note in case we keep this ability)

Removing the site_admin's privileges in this area is by far the simplest solution. Any other would require some determination about a middle ground, only allowing certain roles to be selected/edited for example.

**Comment by zerocrates** [comment]
On the topic of whether it's reasonable to just only allow global admin to create and edit users, I'd appreciate input from @sheilabrennan and @sharonmleon

**Comment by sheilabrennan** [comment]
I think the site admins will need the ability to create users for their sites. 
If not, I think user management may become unwieldy for global admins if, for example, S is deployed across a campus. Unless, I'm misunderstanding the roles. 

Is it possible for site admins to create and edit users with permission level equal and more restrictive than themselves?

**Comment by zerocrates** [comment]
It's possible to restrict which roles can be edited by other roles, but that will add a not insignificant amount of complication, because that won't be a simple "yes/no" check of the type the permissions system is designed around. We'll also have to make sure we're strictly defining a hierarchy of what roles are "lower" than other roles so we know which ones are allowed.

There's also some thinking to be done there: if the idea is that the site admins should be mainly creating users, does it follow that they should be able to edit *all* users of lower roles? It's also not as clear to me that a site admin should be able to create users of *equal* level, that is, more site admins. Spreading around administrator privileges seems like the kind of thing that we might want to be fairly restrictive about.

Basically, the downside I'm focused on is that we have to do much more detailed thinking and deciding of how the roles and creation/update permissions for users should work once we reach down beyond the global level. We'd have to get those decisions nailed down in fairly short order, and then think about actually implementing the setup we need to apply those permission rules.
